### PR TITLE
11 use case 11 vote on solution document upvotedownvote

### PR DESCRIPTION
**Pull request by pandaponn**
Implemented the following according to CRC model:
`VoteOnDocSolutionController`
`VoteOnSolutionDocInteractor`
`VoteSDocDsGateway`
`VoteSDocDsRequestModel`
`VoteSDocInputBoundary`
`VoteSDocOutputBoundary`
`VoteSDocRequestModel`
`VoteSDocResponseModel`

Implemented `getVoteTotalBySolutionIdQuery` and `updateSolutionDocVote` methods in `DatabaseAccessGateway` and `PostgressAccessManager` to enable upvoting/downvoting of solution documents based on solutionId.

**CLARIFICATION:**
According to the description in the issue, when a user clicks on the thumbs up, a boolean value of `true` will be sent as a parameter into the `voteOnSolutionDoc` method. I assumed that `false` would indicate a downvote. As such, the `vote` parameter in `VoteSDocRequestModel` is a boolean to store `true` or `false` so we can identify if it is an upvote or downvote.

However, I have stored the `vote` parameter in `VoteSDocDsRequestModel` as an `int` as this stores the total number of votes to be updated in the DB. I used the same parameter name to conform with the CRC model but this might be confusing due to the way I've implemented it.

### CODE REVIEW

File: src/main/java/fworks/da/PostgresAccessManager.java [hunk 1]
```diff
@@ -672,4 +672,41 @@ public boolean verifyLoginCredentialsQuery(
...
+    }
+
+    /** Queries postgres DB to update the total votes of a solution document
+     *
+     * @param solutionId             the unique solution Id of the solution document being queried
+     * @param voteTotal              the total number of votes to update the solution document to
+     *
+     * @return whether the update of total votes was successful
+     */
+    public boolean updateSolutionDocVote(String solutionId, int voteTotal){
```
Comment (HarveyRDonnelly) [inline src/main/java/fworks/da/PostgresAccessManager.java:702]: I believe there is a typo here and this method should be named updateSolutionDocVoteQuery

File: src/main/java/uc/doc/voteonsolution/VoteSDocResponseModel.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        this.voteTotal = voteTotal;
+        this.solutionDoc = solutionDoc;
+        this.timestamp = timestamp;
+    }
+
+    public int getVoteTotal() {
+        return this.voteTotal;
+    }
+
+    public SolutionDocument getSolutionDoc() {
```
Comment (HarveyRDonnelly) [inline src/main/java/uc/doc/voteonsolution/VoteSDocResponseModel.java:22]: Response models should not depend (or at least return) entities. All response models need rewritten anyway so don't worry about correcting in this PR

**Review by HarveyRDonnelly — CHANGES_REQUESTED** [review]
Looks Great @pandaponn ! The only issues is the typo in PostgresAccessManager that needs resolved

File: src/main/java/fworks/da/PostgresAccessManager.java (hunk 1 above)
Comment (pandaponn) [reply src/main/java/fworks/da/PostgresAccessManager.java]: Great catch @HarveyRDonnelly !

File: src/main/java/fworks/da/PostgresAccessManager.java [hunk 3]
```diff
@@ -672,4 +672,43 @@ public boolean verifyLoginCredentialsQuery(
...
+
+    /** Queries postgres DB to update the total votes of a solution document
+     *
+     * @param solutionId             the unique solution Id of the solution document being queried
+     * @param voteTotal              the total number of votes to update the solution document to
+     *
+     * @return whether the update of total votes was successful
+     */
+    @Override
+    public boolean updateSolutionDocVoteQuery(String solutionId, int voteTotal){
```
Comment (pandaponn) [inline src/main/java/fworks/da/PostgresAccessManager.java:704]: Fixed typo @HarveyRDonnelly

Comment (HarveyRDonnelly) [reply src/main/java/fworks/da/PostgresAccessManager.java:704]: Perfect thank you ! @pandaponn

**Review by HarveyRDonnelly — APPROVED** [review]
All requested changes have been made and use case looks great !

**Pull request merged**